### ISSUE TITLE
GCN loss

### ISSUE THREAD

**Issue by atinyboy** [dinobby/MAGDi#5]
Thank you for the clarification. However, I still have some confusion regarding whether the GCN objective can actually update the decoder parameters in the current implementation.

From the released code, the node embeddings used as GCN inputs appear to be precomputed offline in `get_node_emb.py` using:

```python
with torch.no_grad():
    outputs = model(...)
```

followed by:

```python
.detach().cpu().numpy()
```

These embeddings are then saved and later loaded during MAGDI training as static graph features (`graph_batch.x`).

Because of this, it seems that the computational graph between the GCN loss (`node_loss`) and the decoder/LoRA parameters is disconnected. In other words, the GCN objective appears to only optimize the GCN parameters themselves, while the decoder is updated only through the language modeling loss and margin ranking loss.

If this understanding is correct, then the graph structure supervision may not actually be distilled into the student LLM in an end-to-end manner.

Could you clarify whether:

1. the released implementation intentionally uses detached node embeddings for efficiency reasons, or
2. there exists an online / end-to-end version where the GCN loss backpropagates into the decoder hidden states?

I would really appreciate further clarification on this point, since it seems important for understanding how the structural distillation is actually achieved.
